src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -74,7 +74,11 @@ parseDataDeclaration = do
   tyArgs <- many (indented *> kindedIdent)
   ctors <- P.option [] $ do
     indented *> equals
-    P.sepBy1 ((,) <$> properName <*> P.many (indented *> noWildcards parseTypeAtom)) pipe
+    flip P.sepBy1 pipe $ do
+      dname <- properName
+      tys <- P.many (indented *> noWildcards parseTypeAtom)
+      let fields = [Ident ("value" ++ show n) | n <- [0..(length tys - 1)]]
```

Comment:
Can just be lazy here: `[0..]` and then even move this out into a CAF at the top level to be reused.